src/topology/algebra/continuous_monoid_hom.lean

```diff
@@ -14,54 +14,96 @@ This file defines the space of continuous homomorphisms between two topological
 ## Main definitions
 
 * `continuous_monoid_hom A B`: The continuous homomorphisms `A →* B`.
-
+* `continuous_add_monoid_hom α β`: The continuous additive homomorphisms `α →+ β`.
 -/
 
-variables (A B C D E : Type*)
+open function
+
+variables {F α β : Type*} (A B C D E : Type*)
   [monoid A] [monoid B] [monoid C] [monoid D] [comm_group E]
   [topological_space A] [topological_space B] [topological_space C] [topological_space D]
   [topological_space E] [topological_group E]
 
-set_option old_structure_cmd true
+/-- The type of continuous additive monoid homomorphisms from `α` to `β`.
 
-/-- Continuous homomorphisms between two topological groups. -/
-structure continuous_monoid_hom extends A →* B, continuous_map A B
+When possible, instead of parametrizing results over `(f : continuous_add_monoid_hom α β)`,
+you should parametrize over `(F : Type*) [continuous_add_monoid_hom_class F α β] (f : F)`.
 
-/-- Continuous homomorphisms between two topological groups. -/
+When you extend this structure, make sure to extend `continuous_add_monoid_hom_class`. -/
 structure continuous_add_monoid_hom (A B : Type*) [add_monoid A] [add_monoid B]
-  [topological_space A] [topological_space B] extends A →+ B, continuous_map A B
+  [topological_space A] [topological_space B] extends A →+ B :=
+(continuous_to_fun : continuous to_fun)
+
+/-- The type of continuous monoid homomorphisms from `α` to `β`.
+
+When possible, instead of parametrizing results over `(f : continuous_monoid_hom α β)`,
+you should parametrize over `(F : Type*) [continuous_monoid_hom_class F α β] (f : F)`.
 
-attribute [to_additive] continuous_monoid_hom
-attribute [to_additive] continuous_monoid_hom.to_monoid_hom
+When you extend this structure, make sure to extend `continuous_add_monoid_hom_class`. -/
+@[to_additive]
+structure continuous_monoid_hom extends A →* B :=
+(continuous_to_fun : continuous to_fun)
 
-initialize_simps_projections continuous_monoid_hom (to_fun → apply)
```

Comment:
Could you re-add this line? Or does `simps` generate these projections automatically from an `extends` clause?

Reply:
See [here](https://github.com/leanprover-community/mathlib/pull/11909#discussion_r808937354)